### ISSUE TITLE
exported comment does not work in env:node

### ISSUE THREAD

**Issue by kotodamagic** [eslint/eslint#3833]
## Environment

* Same error is displayed in:
    * node.js@4.0.0 and eslint@1.4.3
    * node.js@4.0.0 and eslint@1.2.0
    * node.js@4.0.0 and eslint@1.4.1
    * node.js@0.12.7 and eslint@1.4.1

## Minimal code to reproduce

.eslintrc:

```json
{
  "root":true,
  "rules": { "no-unused-vars": 2 },
  "env": { "node": true }
}
```

code:

```javascript
/* exported thisIsExported */
var thisIsExported = 0;
```

result:

```
$ eslint no-unused.js

/home/koto/tmp/ruletest/no-unused.js
  2:5  error  "thisIsExported" is defined but never used  no-unused-vars

✖ 1 problem (1 error, 0 warnings)
```

## Issue

* If .eslintrc doesn't contain `"env": { "node": true }`, no errors are displayed.

**Comment by IanVS** [comment]
In node, variables are not added to the global scope, so `thisIsExported` would be only at the module scope.  Since it is not being used in the module, the error is correct.

**Comment by IanVS** [comment]
The docs currently say:

> In some environments you may use var to create a global variable that may be used by other scripts.

Perhaps it would be more explicit to say:

> In some environments **(other than `node` and `commonjs`)** you may use var to create a global variable that may be used by other scripts.
